src/realm/util/network.cpp

```diff
@@ -1155,8 +1155,13 @@ bool Service::IoReactor::wait_and_advance(clock::time_point timeout, clock::time
                     ret++;
                 }
 
-            } while (ret == 0 &&
-                     (duration_cast<milliseconds>(steady_clock::now() - started).count() < max_wait_millis));
+                if (ret != 0 ||
+                    (duration_cast<milliseconds>(steady_clock::now() - started).count() >= max_wait_millis)) {
+                    break;
+                }
+
+                std::this_thread::sleep_for(std::chrono::milliseconds(10));
```

Comment:
can we add a comment about why we do this and make this std::chrono::milliseconds(socket_poll_timeout) rather than a hardcoded 10 milliseconds?

Reply:
Good call. Added one that captures my limited understanding of the problem - let me know if you think that's enough or it needs more details/nuance.